docs/advanced-features/react-18/streaming.md

```diff
@@ -0,0 +1,62 @@
+# Streaming SSR (Alpha)
+
+React 18 will include architectural improvements to React server-side rendering (SSR) performance. This means you could use `Suspense` in your React components in streaming SSR mode and React will send render them and send through HTTP streams. React server components is also an experimental feature based on streaming. Next.js will also provide server components related features along with other streaming support through `next/streaming`.
+
+## Enable Streaming SSR
+
+To enable, use the experimental flag `concurrentFeatures: true`:
+
+```jsx
+// next.config.js
+module.exports = {
+  experimental: {
+    concurrentFeatures: true,
+  },
+}
+```
+
+## Streaming Features
+
+### next/dynamic
+
+Using Dynamic import on React components by `React.lazy` have better support in React 18. Previously Next.js support dynamic import internally without require `Suspense` or `React.lazy`, Now to embrace the official APIs on React side we provide you with `options.suspense` in `next/dynamic`.
+
+```jsx
+import dynamic from 'next/dynamic'
+import { lazy, Suspense } from 'react'
+
+import Content from '../components/content'
+
+// These two ways are identical:
+const Profile = dynamic(() => import('./profile'), { suspense: true })
+const Footer = lazy(() => import('./footer'))
+
+export default function Home() {
+  return (
+    <div>
+      <Suspense fallback={<Spinner />}>
+        {/* A component that uses Suspense-based */}
+        <Content />
+      </Suspense>
+      <Suspense fallback={<Spinner />}>
+        <Profile />
+      </Suspense>
+      <Suspense fallback={<Spinner />}>
+        <Footer />
+      </Suspense>
+    </div>
+  )
+}
+```
+
+Checkout [next/streaming](/docs/api-reference/next/streaming.md) for more details for building Next.js apps in streaming SSR mode.
+
+## Notice
+
+#### Data Fetching
+
+Currently data fetching within suspense boundaries on server side is not fully supported, which could lead to mismatching between server and client. In the short-term, please don't put data fetching within suspense.
```

Comment:
I think this could also use an example of what you mean specifically by data-fetching (e.g. patterns like this won't work...)

```suggestion
Currently, data fetching within Suspense boundaries on the server side is not fully supported, which could lead to mismatching between server and client. In the short-term, please don't try data fetching within Suspense.
```